src/app/components/Arbitrage/Arbitrage.tsx

```diff
@@ -1,77 +1,120 @@
-import React, { useState, useEffect } from 'react';
-import { backendUrl, currentChainId } from '../../../utils/classifiers';
-import axios from 'axios';
-import { symbolByTokenAddress } from 'utils/blockchain/contract-helpers';
+import React, { useMemo } from 'react';
+import { bignumber } from 'mathjs';
 import { useTranslation } from 'react-i18next';
 import { translations } from 'locales/i18n';
-import { Popover, Icon } from '@blueprintjs/core';
+import { Icon, Popover } from '@blueprintjs/core';
+import { useSelector } from 'react-redux';
+import { assetByTokenAddress } from 'utils/blockchain/contract-helpers';
+import { backendUrl, currentChainId } from '../../../utils/classifiers';
+import { useFetch } from '../../hooks/useFetch';
+import { Asset } from '../../../types';
+import { selectWalletProvider } from '../../containers/WalletProvider/selectors';
+import { fixNumber } from '../../../utils/helpers';
+import { AssetSymbolRenderer } from '../AssetSymbolRenderer';
+import { toNumberFormat } from '../../../utils/display-text/format';
 
 const s = translations.swapTradeForm;
 
+const minUsdForOpportunity = 10;
+
+interface PoolData {
+  [pool: string]: {
+    oracleRate: string;
+    negativeDelta: boolean;
+    rateToBalance: {
+      amount: number;
+      from: string;
+      to: string;
+      rate: number;
+      earn: number;
+    };
+  };
+}
+
 export function Arbitrage() {
   const { t } = useTranslation();
-  const api = backendUrl[currentChainId];
-  const [show, setShow] = useState(false);
-  const [data, setData] = useState({
-    USDT: {
-      oracleRate: '0',
-      negativeDelta: false,
-      rateToBalance: { amount: 0, from: '', to: '', rate: 0, earn: 0 },
-    },
-  });
+  const { assetRates } = useSelector(selectWalletProvider);
 
-  useEffect(() => {
-    axios
-      .get(api + '/amm/arbitrage')
-      .then(res => setData(res.data))
-      .catch(console.error);
-  }, [api]);
+  const { value: data } = useFetch<PoolData>(
```

Comment:
I am getting CORS errors on local so I couldn't test it with real data but we can test it on testnet in the worst-case scenario.

Reply:
Maybe backend was down or dev branch uses wrong backend in the classifiers (should be https://api.test.sovryn.app)

Reply:
Probably the backend was down, I tested it today and it worked like a charm.